stan/math/rev/arr/functor/coupled_ode_system.hpp

```diff
@@ -38,14 +38,32 @@ namespace math {
  * <p>The final M states correspond to the sensitivities with respect
  * to the second base system equation, etc.
  *
+ * <p>Note: For efficiency reasons the parameter vector is used as
```

Comment:
I think this doc doesn't describe the conditions clearly.

Shouldn't this state that nothing else can touch the autodiff stack for the duration of this object? We need to start being careful about this because we're moving to parallel computation. (This particular condition requires dealing with this object with care and the user should be trying to limit the lifetime of the objects as much as possible.)